lib/market.rb

```diff
@@ -1,3 +1,80 @@
 class Market
- 
+require 'date'
+    
+    attr_reader :name,
+                :date
+
+    attr_accessor :vendors
+
+    def initialize(name)
+        @name = name
+        @vendors = []
+        @date = Date.today.strftime('%Y/%m/%d')
+    end
+
+    def add_vendor(vendor_object)
+        @vendors << vendor_object
+    end
+
+    def vendor_names
+        @vendors.map do |vendor|
+            vendor.name
+        end
+    end
+
+    def vendors_that_sell(item)
+        @vendors.select do |vendor|
+            vendor.check_stock(item) > 0 
+        end
+    end
+
+    def sorted_item_list
+        all_items = @vendors.flat_map do |vendor|
+            vendor.inventory.keys
+        end
+
+        all_items.uniq.sort_by do |item| 
+            item.name 
+        end
+    end
+
+    def total_inventory
+        total_inventory = {}
+        
+        @vendors.each do |vendor|
+            vendor.inventory.each do |item, quantity|
+
+                if total_inventory[item]
+                    total_inventory[item][:quantity] += quantity
+                    total_inventory[item][:vendors] << vendor unless total_inventory[item][:vendors].include?(vendor)
```

Comment:
There may be an easier way to do it! Might be worth asking chatgpt for some refactor tips.